echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/TriggerConverter.groovy

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.scheduler.actions.pipeline
+
+import com.netflix.spinnaker.echo.cron.CronExpressionFuzzer
+import com.netflix.spinnaker.echo.model.Pipeline
+import com.netflix.spinnaker.echo.model.Trigger
+import com.netflix.spinnaker.echo.pipelinetriggers.PipelineCache
+import org.quartz.CronExpression
+import org.quartz.JobDataMap
+
+import static org.quartz.CronScheduleBuilder.cronSchedule
+import static org.quartz.TriggerBuilder.newTrigger
+
+class TriggerConverter {
+  public static final String JOB_ID = "Pipeline Trigger"
+
+  static Map<String, String> toParamMap(Trigger trigger, String timeZoneId) {
+    def params = [
+      id                   : trigger.parent.id,
+      application          : trigger.parent.application,
+      triggerId            : trigger.id,
+      triggerType          : Trigger.Type.CRON.toString(),
+      triggerCronExpression: trigger.cronExpression,
+      triggerTimeZoneId    : timeZoneId,
+      triggerRebake        : Boolean.toString(trigger.rebake),
+      triggerEnabled       : "true"
+    ]
+
+    if (trigger.runAsUser) {
+      params.runAsUser = trigger.runAsUser
+    }
+
+    return params
+  }
+
+  static org.quartz.Trigger toQuartzTrigger(Trigger pipelineTrigger, TimeZone timeZoneId) {
+    if (pipelineTrigger.cronExpression == null) {
+      throw new InvalidCronExpressionException("<NULL>")
```

Comment:
I'd make this exception more meaningful.

Reply:
`InvalidCronExpressionException` will already expand the text to
`Invalid CRON expression: ${cronExpression}`. Do we need anything more than that?

Reply:
Ah, nice. Perhaps including the pipelineConfigId if available?

Reply:
I find the syntax a bit weird. If the error is that the thing is `null`, can we just say it's `null` and not `<NULL>`?

Reply:
sure. the pipeline ID will already be spit out because this exception is always caught